provider/internal/builtin/service_client.go

```diff
@@ -280,30 +240,3 @@ func (p *builtintServiceClient) Evaluate(cap string, conditionInfo []byte) (prov
 		return response, fmt.Errorf("capability must be one of %v, not %s", capabilities, cap)
 	}
 }
-
-func findFilesMatchingPattern(root, pattern string) ([]string, error) {
-	var regex *regexp.Regexp
-	// if the regex doesn't compile, we'll default to using filepath.Match on the pattern directly
-	regex, _ = regexp.Compile(pattern)
-	matches := []string{}
-	err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
-		var matched bool
-		if regex != nil {
-			matched = regex.MatchString(d.Name())
-		} else {
-			// TODO(fabianvf): is a fileglob style pattern sufficient or do we need regexes?
-			matched, err = filepath.Match(pattern, d.Name())
```

Comment:
I think the error in the demo output is coming from the removal of this branch, `*.go` is a fileglob style pattern instead of regex and so will not compile.

Reply:
implemented